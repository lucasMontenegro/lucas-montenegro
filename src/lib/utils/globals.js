import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g