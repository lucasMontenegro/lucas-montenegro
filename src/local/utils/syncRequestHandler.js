const { console } = global
module.exports = function syncRequestHandler (asyncHandler) {
  return (req, res, next = console.error) => Promise.resolve(asyncHandler(req, res)).catch(next)
}