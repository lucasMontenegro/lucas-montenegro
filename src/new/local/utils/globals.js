// renaming "global" to "globals"
// declaring a variable named "global" may cause troubles
exports.default = require("global")