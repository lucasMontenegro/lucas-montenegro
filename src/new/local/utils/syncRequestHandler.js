import globals from "new/local/utils/globals"
export default function syncRequestHandler (asyncHandler) {
  return (req, res, next = globals.console.error) => {
    return Promise.resolve(asyncHandler(req, res)).catch(next)
  }
}