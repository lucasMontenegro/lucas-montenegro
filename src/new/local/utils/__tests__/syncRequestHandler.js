import globals from "new/local/utils/globals"
import syncRequestHandler from "new/local/utils/syncRequestHandler"
jest.mock(`new/local/utils/globals`, () => ({
  __esModule: true,
  default: { console: { error: jest.fn() } },
}))
describe(`local/utils/syncRequestHandler`, () => {
  it(`should resolve promises`, async () => {
    const result = {}
    const asyncFn = async (req, res) => result
    expect(await syncRequestHandler(asyncFn)(`req`, `res`, () => {})).toBe(result)
  })
  {
    const error = Error(`foo`)
    const throwAsyncError = async (req, res) => {
      throw error
    }
    function expectToThrow (fn) {
      expect(fn.mock.calls).toHaveLength(1)
      expect(fn.mock.calls[0]).toHaveLength(1)
      expect(fn.mock.calls[0][0]).toBe(error)
    }
    it(`should catch errors`, async () => {
      const next = jest.fn()
      await syncRequestHandler(throwAsyncError)(`req`, `res`, next)
      expectToThrow(next)
    })
    it(`should use 'console.error' as the default error handler`, async () => {
      await syncRequestHandler(throwAsyncError)(`req`, `res`)
      expectToThrow(globals.console.error)
    })
  }
})