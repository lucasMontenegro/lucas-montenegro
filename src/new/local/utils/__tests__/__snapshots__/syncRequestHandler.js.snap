// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local/utils/syncRequestHandler should utilize the dependency APIs correctly 1`] = `
Object {
  "express": "^4.17.1",
}
`;
